core/src/main/scala/org/apache/spark/internal/config/package.scala

```diff
@@ -228,6 +228,16 @@ package object config {
     .timeConf(TimeUnit.SECONDS)
     .createWithDefaultString("1m")
 
+  private[spark] val KERBEROS_RENEWAL_CREDENTIALS =
+    ConfigBuilder("spark.kerberos.renewal.credentials")
```

Comment:
Wondering why an additional config needed and not just falling back to cache if keytab is not provided + the cluster is secure.

Reply:
That would be a change in behavior. e.g in YARN that would increase the number of delegation tokens an app creates when it's run without token renewal.

Second, if your TGT is not renewable (`kinit -l foo`, instead of `kinit -l foo -r bar`), you'll get noisy errors in the output.

Also, a config opens the path for other ways of providing tokens (e.g. renewing based on a delegation token cache, which some k8s guys were interested in).

Reply:
Now I see, it's fine.

A little bit offtopic here but I've considered to apply this to the kafka area. If no global JVM security config provided then use this config or something.